src/Plugin/GraphQL/DataProducer/User/PasswordReset.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+namespace Drupal\graphql\Plugin\GraphQL\DataProducer\User;
+
+use Drupal\Core\Logger\LoggerChannelInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\graphql\Plugin\GraphQL\DataProducer\DataProducerPluginBase;
+use Drupal\graphql\GraphQL\Response\ViolationResponse;
+use Drupal\graphql\GraphQL\Violation\ViolationCollection;
+use Drupal\user\Controller\UserAuthenticationController;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Resets the user's password (mutation).
+ *
+ * @DataProducer(
+ *   id = "password_reset",
+ *   name = @Translation("Password reset"),
+ *   description = @Translation("Allows to reset the password."),
+ *   consumes = {
+ *     "email" = @ContextDefinition("email",
+ *       label = @Translation("Email")
+ *     )
+ *   }
+ * )
+ */
+class PasswordReset extends DataProducerPluginBase implements ContainerFactoryPluginInterface {
+
+  use StringTranslationTrait;
+
+  /**
+   * List of validation violations.
+   *
+   * @var \Drupal\graphql\GraphQL\Violation\ViolationCollection
+   */
+  protected $violations;
+
+  /**
+   * The current request.
+   *
+   * @var \Symfony\Component\HttpFoundation\Request
+   */
+  protected $currentRequest;
+
+  /**
+   * The logger service.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelInterface
+   */
+  protected $logger;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    /** @var \Symfony\Component\HttpFoundation\RequestStack $request_stack */
+    $request_stack = $container->get('request_stack');
+    /** @var \Drupal\Core\Logger\LoggerChannelInterface $logger */
+    $logger = $container->get('logger.channel.graphql');
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $request_stack->getCurrentRequest(),
+      $logger
+    );
+  }
+
+  /**
+   * UserRegister constructor.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param array $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Symfony\Component\HttpFoundation\Request $current_request
+   *   The current request.
+   * @param \Drupal\Core\Logger\LoggerChannelInterface $logger
+   *   The logger service.
+   */
+  public function __construct(
+    array $configuration,
+    string $plugin_id,
+    array $plugin_definition,
+    Request $current_request,
+    LoggerChannelInterface $logger
+  ) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->currentRequest = $current_request;
+    $this->logger = $logger;
+    $this->violations = new ViolationCollection();
```

Comment:
this can be a local variable in resolve(), we don't need it on the class?

Reply:
done